### PR TITLE
fix(set): support bundled -o in short-flag clusters (set -euo pipefail)

### PR DESCRIPTION
**Pull request by chernetsov**
## Problem

`set -euo pipefail` — the canonical "bash strict mode" idiom — fails:

```
$ echo 'set -euo pipefail; echo ok' | just-bash
bash: set: -o: invalid option
set: usage: set [-eux] [+eux] [-o option] [+o option]
...
```

The cluster branch of `handleSet` parses every character after `-` as an
independent short flag, and there is no `o` short flag. `-o` was only honored
as its own token (`set -eu -o pipefail`). Because `set -euo pipefail` is what
LLMs / scripts put on the *first line*, the whole script aborts before doing
anything. This reproduces on `main` (and every published version through
3.0.1).

## Fix

Match bash's behavior: an `o` encountered inside a short-flag cluster consumes
the **next word** as its long-option name, and the remaining characters of the
token keep being parsed as short flags.

- `set -euo pipefail` ≡ `set -e -u -o pipefail`
- `set -oe pipefail` enables both `pipefail` (from the next word) and `errexit`
  (the trailing `e`) — order within the token doesn't matter, matching bash
- multiple `o`s consume successive words
- trailing words become positional parameters (`set -euo pipefail a b` → `$1=a`, `$2=b`)
- `+`-clusters disable (`set +euo pipefail`)
- an invalid bundled name still errors with `invalid option name`
  (`set -euo bogus`)
- an `o` with no available argument falls back to the standalone `-o`/`+o`
  listing

The two standalone `-o` / `+o` listing branches were extracted into shared
`formatOptionsList` / `formatOptionsResetCommands` helpers and reused by the
cluster fallback (no behavior change for the standalone forms).

## Tests

Added a `bundled -o in a short-flag cluster (strict mode)` describe block in
`set.test.ts` covering `-euo pipefail`, errexit/nounset/pipefail activation,
`-eo pipefail`, the `-oe pipefail` ordering case, trailing positional params,
`+euo pipefail`, and the invalid-name error.

Verified locally:

- `vitest run` on `set.test.ts` (51 passed), `set-pipefail.test.ts` +
  `set-errexit.test.ts` (37 passed)
- `tsc --noEmit` clean
- `biome check` + `lint:banned` clean

Changeset included (`patch`).

### CODE REVIEW

**Review by AGambit95 — APPROVED** [review]

File: packages/just-bash/src/interpreter/builtins/set.ts [hunk 1]
```diff
@@ -368,35 +391,61 @@ export function handleSet(ctx: InterpreterContext, args: string[]): ExecResult {
...
+                throw new PosixFatalError(1, "", errorMsg);
+              }
+              return failure(errorMsg);
+            }
+            setShellOption(ctx, LONG_OPTION_MAP.get(optName) ?? null, enable);
+            consumedArgs++;
+            continue;
+          }
+          // No option name available: behave like a standalone `-o`/`+o`.
+          return success(
```
[the remark this answers is not in the document]
Comment (chernetsov) [reply packages/just-bash/src/interpreter/builtins/set.ts]: Good catch — fixed in 89beabb. A no-option-name `o` in a cluster (e.g. `set -oe`) now keeps processing the remaining flags instead of returning early, so errexit is enabled while the option listing is still printed. Verified against bash: `set -oe` → `$-` = `ehB` (errexit on) plus the listing. Added regression tests for `set -oe`, `set -eo`, and `set -oue`.

File: packages/just-bash/src/interpreter/builtins/set.ts [hunk 2]
```diff
@@ -368,35 +391,61 @@ export function handleSet(ctx: InterpreterContext, args: string[]): ExecResult {
...
+            enable ? formatOptionsList(ctx) : formatOptionsResetCommands(ctx),
+          );
```
[the remark this answers is not in the document]
Comment (chernetsov) [reply packages/just-bash/src/interpreter/builtins/set.ts:470]: This guard is intentional and matches bash. bash does not treat a following `-`/`+`-prefixed token as the `-o` option name — it falls back to listing and then processes that token as its own flag. Verified:

- `set -o -x` → prints the options listing, then enables xtrace (`$-` = `hxB`)
- `set -eo -x` → `$-` = `ehxB` (errexit from `e`, xtrace from the trailing `-x`, listing in between)
- `set +o -x` → prints `set +o ...` reset commands, then enables xtrace

So a name must be a non-option word; otherwise it's the "no argument available" case. This mirrors the standalone `-o` handler, which uses the same `hasNonOptionArg` guard. Consuming `-x` as the option name would diverge from bash. I tightened the comment wording to make this explicit.

File: packages/just-bash/src/interpreter/builtins/set.test.ts [hunk 3]
```diff
@@ -224,6 +224,90 @@ describe("set builtin", () => {
     });
   });
 
+  describe("bundled -o in a short-flag cluster (strict mode)", () => {
+    it("should accept `set -euo pipefail`", async () => {
+      const env = new Bash();
```
[the remark this answers is not in the document]
Comment (chernetsov) [reply packages/just-bash/src/interpreter/builtins/set.test.ts:229]: Added targeted tests for both edge cases:
- No-name listing fallback that still applies the remaining flags: `set -oe`, `set -eo`, `set -oue` (89beabb), plus `set -o -x` (48870ce).
- Multiple `o`s consuming successive words: `set -oo pipefail errexit` (259d78b).

**Comment by chernetsov** [comment]
All commits on this branch are now signed and show as **Verified** on GitHub.

**Review by cramforce — APPROVED** [review]
